synthesis/codon/analyzer.go

```diff
@@ -0,0 +1,36 @@
+package codon
+
+import "github.com/TimothyStiles/poly/io/genbank"
+
+// Stats denotes a set of computed codon table statistics
+type Stats struct {
+	StartCodonCount map[string]int `json:"start_codons_counts"`
+}
+
+func NewStats() *Stats {
+	return &Stats{
+		StartCodonCount: map[string]int{},
+	}
+}
+
+// Analyzer is used to compute codon usage statistics
+type Analyzer struct{}
```

Comment:
Empty struct with a single method could just be a function. Structs are for storing state/bundling data.

Reply:
not always, here's one [in kubernetes](https://github.com/rancher/kubernetes/blob/6668bdc8423b0cc7eae01080fcbddb1837652c74/pkg/routes/ui.go#L28). You use structs like this to make it easier to mock in unit tests, for example we might want to test something which relies on the analyser without actually calling the struct's code.

e.g we can imagine elsewhere: 

```go
type Analyser interface{
	ComputeStats(sequence genbank.Genbank) (*Stats, error)
}


// test.go

type mockAnalyzer struct{}

func (a *Analyzer) ComputeStats(sequence genbank.Genbank) (*Stats, error) {
	return //stats, mockError
}
```

Reply:
This makes sense in the Kubernetes example because these structs implement a common interface. I don't see a use case for defining an interface here.

Getting the statistics from a codon table should be a pure function we can extensively unit test, what scenarios do you have in mind where we would need to mock it?